build-image/Dockerfile

```diff
@@ -0,0 +1,67 @@
+FROM docker.io/bitnami/minideb:bullseye-amd64 as slimmed-rust
+
+# The rust version to use
+# The 1.69 toolchain is temporarily required to build ink! contracts because of
+# https://github.com/paritytech/cargo-contract/issues/1139
+ARG RUST_VERSION=1.69
+# The cargo contract version to use
+ARG CARGO_CONTRACT_VERSION=3.0.1
+
+# metadata
+LABEL io.parity.image.vendor="Parity Technologies" \
+    io.parity.image.title="paritytech/contracts-verifiable" \
+    io.parity.image.description="Inherits from docker.io/bitnami/minideb:bullseye. \
+    rust nightly, clippy, rustfmt, miri, rust-src, rustc-dev, grcov, rust-covfix, \
+    llvm-tools-preview, cargo-contract, xargo, binaryen, parallel, codecov, ink, solang" \
+    io.parity.image.documentation="https://github.com/paritytech/cargo-contract/blob/master/\
+    build-image/README.md" \
+    io.parity.version.rust=${RUST_VERSION} \
+    io.parity.version.cargo-contract=${CARGO_CONTRACT_VERSION}
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH
+
+# Minimal Rust dependencies.
+RUN set -eux \
+    && apt-get update && apt-get -y install wget \
+    && url="https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init" \
+    && wget "$url" \
+    && chmod +x rustup-init \
+    && ./rustup-init -y --no-modify-path --profile minimal --component rust-src rustfmt --default-toolchain $RUST_VERSION  \
+    && rm rustup-init \
+    && chmod -R a+w $RUSTUP_HOME $CARGO_HOME \
+    && rustup --version \
+    && cargo --version \
+    && rustc --version \
+    && apt-get remove -y --auto-remove wget \
+    && apt-get -y install gcc \
+    && rm -rf /var/lib/apt/lists/*
+
+FROM slimmed-rust as cc-builder
+ARG CARGO_CONTRACT_VERSION
+
+# This is important, see https://github.com/rust-lang/docker-rust/issues/85
+ENV RUSTFLAGS="-C target-feature=-crt-static"
+
+    # Install required packages for `cargo-contract`
+RUN apt-get -y update && apt-get -y install gcc g++ \
+    # Install cargo contract
+    && cargo install cargo-contract --locked --version $CARGO_CONTRACT_VERSION \
+    # Cleanup after `cargo install`
+    && rm -rf ${CARGO_HOME}/"registry" ${CARGO_HOME}/"git" /root/.cache/sccache \
+    # apt clean up
+    && apt-get remove -y gnupg \
+    && apt-get autoremove -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+
+FROM slimmed-rust as ink-dev
+
+COPY --from=cc-builder /usr/local/cargo/bin/cargo-contract /usr/local/bin/cargo-contract
+
+WORKDIR /contract
+
+# default entry point
+ENTRYPOINT ["cargo", "contract", "build", "--release"]
```

Comment:
So does this mean `cargo contract build --verifiable` runs the release build by default? I understand that it usually only makes sense to do this for release builds.

However might be a bit unintuitive if the `build` command behaves slightly differently with the `--verifiable` command.

Reply:
The entrypoint is overwritten in rust code actually. However, the behavior is similar: the build will always be executed in release and I believe we should keep it that way since I don't really know the case when you want to produce the verifiable build in debug mode.

My understanding of `--verifiable` is to be an alternative flag to `--release`

Reply:
> I don't really know the case when you want to produce the verifiable build in debug mode

I agree, the reason that I brought it up is that my mental model was that the effect of this command was to execute `build` faithfully in the container with the exact same args.

> My understanding of --verifiable is to be an alternative flag to --release

Okay, then it should be a variant on `BuildMode` to avoid confusion. As it is you could still have `--release` and it would produce the same result with and without that. 

A small con to this would be it is mixing `debug` and `release` cargo profiles with a different concept.